BeeJet.Bot/ClientHandlers/SlashCommandHandler.cs

```diff
@@ -1,32 +1,81 @@
-﻿using BeeJet.Bot.Commands.Sources;
+﻿using BeeJet.Bot.Attributes;
+using BeeJet.Bot.Commands;
+using BeeJet.Bot.Commands.Handlers;
+using BeeJet.Bot.Extensions;
+using Discord;
 using Discord.Commands;
 using Discord.WebSocket;
+using System.Reflection;
 
 namespace BeeJet.Bot.ClientHandlers
 {
     internal class SlashCommandHandler : BaseClientHandler
     {
-        private readonly List<ICommandSource> _commandSources;
+        private readonly List<(Type ClassType, MethodInfo Method, string CommandName)> _commandMethods;
 
-        public SlashCommandHandler(DiscordSocketClient client, CommandService service, IServiceProvider serviceProvider, List<ICommandSource> commandSources)
+        public SlashCommandHandler(DiscordSocketClient client, CommandService service, IServiceProvider serviceProvider)
             : base(client, service, serviceProvider)
         {
-            _commandSources = commandSources;
+            _commandMethods = GetCommandMethods();
+        }
+        private List<(Type ClassType, MethodInfo Method, string CommandName)> GetCommandMethods()
+        {
+            var commandSources = GetCommandSourceTypes();
+            return commandSources.SelectMany(c => c.GetMethods()
```

Comment:
Ik ben hier zelf nooit echt fan van. Ale die 1 character dingen. 
c en m zeggen niets.. totdat je het helemaal gaat lezen en ziet dat c een command en m blijkbaar een method is.